opengovsg/plumber
# PLU-81: [IF-THEN-14] Implement branch deletion

## Problem
We need to support deleting if-then branches.

## Solution
This PR implements a "delete branch" button. It also pops up a warning / confirmation modal before deletion, since deleting the branch means deletion of multiple steps.

<img width=500 src="https://github.com/opengovsg/plumber/assets/135598754/2b25676f-a406-413a-89b4-70f5f0823da3" />

#### Small RFC
Do we need the warning modal for branches without any steps? Also pending PD inputs on the copy.

## Tests
- Check that branch is not deleted if user clicks on "cancel" in the confirmation modal.
- Check that deleting a branch deletes all steps in that branch as well.
- Regression test: check that deleting a step that is outside if-then still works.
